### PR TITLE
Fi bo 339 metrics refactoring

### PR DESCRIPTION
**Pull request by Cebox82**

### CODE REVIEW

File: android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragment.java [hunk 1]
```diff
@@ -116,6 +123,26 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+                .setTitleText("Select a time")
+                .build();
+        timePicker.addOnPositiveButtonClickListener(dialog -> {
+            hours = timePicker.getHour();
+            minutes = timePicker.getMinute();
+            dateText.append(", "+hours+":"+minutes+" Uhr");
+
+        });
+    }
+
```
Comment (Jonas-Karl) [inline android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragment.java:145]: I think you mistakenly added the timepicker to this branch

File: android-app/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
     <string name="adding_expense">Ausgabe</string>
     <string name="adding_income">Einnahme</string>
     <string name="adding_amount">Betrag (€)</string>
-    <string name="adding_date">Datum</string>
+    <string name="adding_date">Zeitstempel</string>
```
Comment (Jonas-Karl) [inline android-app/app/src/main/res/values/strings.xml:32]: This is a change made due to the introduction of the timepicker

File: android-app/app/src/main/res/layout/fragment_adding.xml [hunk 3]
```diff
@@ -99,7 +99,7 @@
                     android:layout_weight="0" />
 
                 <com.google.android.material.textfield.TextInputLayout
-                    android:layout_weight="1"
+                    android:layout_weight="2"
```
Comment (Jonas-Karl) [inline android-app/app/src/main/res/layout/fragment_adding.xml:102]: This is a change made due to the introduction of the timepicker

File: android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/MaterialDatePickerMatcher.java
Comment (Jonas-Karl) [inline android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/MaterialDatePickerMatcher.java:1]: It looks like you dont need this file anymore

File: android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java
Comment (Jonas-Karl) [inline android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java:1]: It looks like every change in this file is also part of the time picker introduction. You can revert these changes as you already managed to test the time picker in #80

File: android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/TestMatchers.java [hunk 4]
```diff
@@ -1,13 +1,35 @@
...
+            }
+
+            @Override
+            public boolean matchesSafely(View view) {
+                ViewParent parent = view.getParent();
+                return parent instanceof ViewGroup && parentMatcher.matches(parent)
+                        && view.equals(((ViewGroup) parent).getChildAt(position));
+            }
+        };
+    }
```
Comment (Jonas-Karl) [inline android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/TestMatchers.java:32]: This should also not be a part of this branch

File: android-app/app/build.gradle [hunk 5]
```diff
@@ -69,6 +69,7 @@ dependencies {
     implementation 'androidx.navigation:navigation-ui:2.5.3'
     implementation 'androidx.recyclerview:recyclerview:1.3.0'
     implementation 'com.github.PhilJay:MPAndroidChart:v3.1.0'
+    implementation 'androidx.test.espresso:espresso-contrib:3.5.1'
```
Comment (Jonas-Karl) [inline android-app/app/build.gradle:72]:
```suggestion
```

I guess you dont need to introduce this dependency for the refactoring

**Review by Jonas-Karl — CHANGES_REQUESTED** [review]
You added many things that were probably mistakenly introduced in this branch. Please revert them. I think the rest looks fine :)

**Review by Jonas-Karl — APPROVED** [review]

**Comment by Jonas-Karl** [comment]
There are still some code smells that can be addressed easily. Also you have to fix the conflics with the master branch and you should check why the android CI is failing. I guess you just need to merge the master again, but  did not test it, that's why I'm not sure :)

File: android-app/app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -11,7 +11,7 @@
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher"
         android:supportsRtl="true"
-        android:usesCleartextTraffic="true"
+        android:usesCleartextTraffic="false"
```
Comment (Jonas-Karl) [inline android-app/app/src/main/AndroidManifest.xml:14]: I dont think this works. Actually, we need clear traffic to be enabled when we are developing, because we use `http` instead of `https`. I think we need to do this change as soon as we are building the production app. In that case we should also change the android:debuggable to `false`. Anyways, I think its too early for us to change this value, as we are still in development. In production it will work without clear traffic, because we will have an `https` protocol

**Comment by jens1o** [comment]
please either a) fix the merge conflicts or b) redo the changes based on a current master.

Looking at the conflict, I'd choose option b).

**Review by jens1o — APPROVED** [review]
